### PR TITLE
DanBeauregard - simple gke blueprint

### PR DESCRIPTION
**Pull request by danbeauregard**
New simple blueprint to create a gke cluster in Google Cloud.

## Definition of Done

This is the definition of done for (new and modified) blueprints to be accepted into the [central XebiaLabs blueprints repository](https://github.com/xebialabs/blueprints).

### Value

This section describes how to determine whether a blueprint has enough value to be included in the central XebiaLabs blueprints repository.

- [x] The blueprint applies to a focused use case.
- [x] The blueprint uses at least one XebiaLabs product.
- [x] The blueprint depends on as few other products as possible. **N.B.:** This is part of the definition of done to keep the blueprint focused and applicable to as many users as possible. A blueprint that depends on one specific CI tool **and** one specific Java EE application server **and** one specific ticketing system can only be used by users that have all three products.
- [x] The blueprint can be used with content supplied by the user but also includes demo content. A blueprint that does not work with content supplied by the user is a demo or a showcase and does not help the user with their own work.

### Technical

This section describes how to determine whether the blueprint has the right technical quality to be included in the central XebiaLabs blueprints repository.

- [x] The branch from which the PR is created is up-to-date with the `development` branch.
- [x] The blueprint contains a `README.md` file at the root of the blueprint folder that describes the blueprint, using the README template in the `.github` folder of this repository.
- [x] The blueprint generates a `xebialabs/USAGE.md` file which, at a minimum, explains how to use the blueprint after it is generated (like adding any missing steps, creating accounts, setting up Docker containers, applying the YAML using `xl` CLI, running release. etc.). **N.B.:** Do not use this document to describe how to instantiate the blueprint. It will only be available to the user *after* the blueprint has been instantiated.
- [x] The blueprint generates a `docker/docker-compose.yml` file which can be used to spin up the required products that the blueprints uses. This can be used to test and/or try out the blueprint without having to manually install those products.
- [x] The blueprint does not contain sensitive information such passwords, tokens, credentials or licenses.
- [x] The blueprint uses `secret: true` in the `blueprint.yaml` parameter definition for question that ask for sensitive information.
- [x] The blueprint does not contain the files `xebialabs/.gitignore`, `xebialabs/values.xlvals` and `xebialabs/secrets.xlvals` and does not refer to them from the files section of the `blueprint.yaml` file. These files will be generated when the blueprint is instantiated. To generate the `xebialabs/values.xlvals` file, use the `saveInXlVals: true` directive in the parameters section of the blueprint. To generate the `xebialabs/secrets.xlvals` file, use the `secret: true` directive .
- [x] The blueprint does not define parameters for trivial things like phase names, task names, folder names etc. Ask questions that add value and be opinionated where possible. For example, ask for a project name and derive folder names, task names etc from that.
- [x] Folder and file names must use [kebab-case](http://wiki.c2.com/?KebabCase), for example: `aws/sample-app-demo`, `xld-environment.yaml`

### Review and testing

This section describes how to determine whether the blueprint has been reviewed and tested well enough to be included in the central XebiaLabs blueprints repository.

- [x] Unit test is added in a `__test__` folder for each blueprint. Refer the `CONTRIBUTING.md` file at the root of of this repository for more details.
- [x] The Travis CI for the PR is green
- [x] The blueprint has been reviewed by someone else in the team.
- [x] Both README and USAGE files have been reviewed by a technical writer and a product marketing manager.
- [x] The blueprint has been manually tested with the `docker/docker-compose.yml` that is generated as part of it.
- [x] The blueprint works on Linux, Mac and Windows.

### CODE REVIEW

File: gcp/gke/xebialabs/USAGE.md.tmpl [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+2. Go to the UI for XL Release (http://localhost:5516), look for the "{{$cluster}}-create" template, and start a new release from it.
+3. Once you are done, go to XL Release and look for the "{{$cluster}}-destroy" template. Start a new release from it to automatically rollback and destroy the resources you created.
+
+## Minimum required versions
+
+This blueprint version requires at least the below tool versions to work properly.
+
+XL Release: Version 8.6.1
+XL Deploy: Version 8.6.1
+XL CLI: Version 8.6
```
Comment (erincinci) [inline gcp/gke/xebialabs/USAGE.md.tmpl]: Should be 8.6.1 as well

File: gcp/gke/xebialabs/xlr-pipeline-create.yaml.tmpl [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+apiVersion: xl-release/v1
+kind: Templates
+spec:
+- name: {{$cluster}}
+  type: xlrelease.Folder
+  children:
+  - name: {{$cluster}}-create
+    type: xlrelease.Release
+    description: |
+      This XL Release template shows how to deploy an application, based on microservices architecture, to GCP GKE using XL Deploy and Terraform.
```
Comment (erincinci) [inline gcp/gke/xebialabs/xlr-pipeline-create.yaml.tmpl]: Leftover microservice related description text?

File: gcp/gke/xebialabs/xlr-pipeline-destroy.yaml.tmpl [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+apiVersion: xl-release/v1
+kind: Templates
+spec:
+- name: {{$cluster}}
+  type: xlrelease.Folder
+  children:
+  - name: {{$cluster}}-destroy
+    type: xlrelease.Release
+    description: |
+      This XL Release template shows how to undeploy an application, based on microservices architecture, to GCP GKE using XL Deploy and Terraform.
```
Comment (erincinci) [inline gcp/gke/xebialabs/xlr-pipeline-destroy.yaml.tmpl]: Same comment as above, leftover microservice ..

**Comment by vpartington** [comment]
Hi Dan,

I will close this PR. The [azure/basic-aks-cluster](https://github.com/xebialabs/blueprints/tree/development/azure/basic-aks-cluster) blueprint now addresses this use case.

If you have any further modifications to make, please create a PR based on that blueprint.

Regards, Vincent.

**Comment by danbeauregard** [comment]
I think there is value to having a simple blueprint available in all clouds.  We can't assume everyone has access to azure.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by vpartington** [comment]
I made a copy/paste error. The blueprint I wanted to refer to is the `gcp/basic-gke-cluster` one.

**Pull request closed without merging**